### PR TITLE
Adding Github and LinkedIn prod guide documentation for Java SDK

### PR DESCRIPTION
**Pull request by mrioan**

### CODE REVIEW

File: source/java/product-guide/index.markdown [hunk 1]
```diff
@@ -3721,6 +3721,218 @@ The returned `FacebookProviderData` will include:
 
 ***
 
+<a class="anchor" name="integration-github"></a>
+## Integrating with Github
+
+Stormpath supports accessing accounts from a number of different locations including Github.  Github uses OAuth 2.0 protocol for authentication / authorization and Stormpath can leverage their or access tokens to return an `Account` for a given code.
```
Comment (maranai) [inline source/java/product-guide/index.markdown]: Not sure what you mean by "Stormpath can leverage their or access tokens"

File: source/java/product-guide/index.markdown [hunk 2]
```diff
@@ -3721,6 +3721,218 @@ The returned `FacebookProviderData` will include:
...
+Creating a Github Directory is very similar to [creating a directory](#create-a-directory) within Stormpath.  For a Github Directory to be configured correctly, you must specify the correct `Provider` information. 
+
+**Example Request**
+
+    Directory directory = client.instantiate(Directory.class);
+    directory.setName("my-github-directory")
+             .setDescription("A Github directory");
+
+    CreateDirectoryRequest request = Directories.newCreateRequestFor(directory).
+                forProvider(Providers.GITHUB.builder()
```
Comment (maranai) [inline source/java/product-guide/index.markdown]: .forProvider (move here the . from the previous line)

File: source/java/product-guide/index.markdown [hunk 3]
```diff
@@ -3721,6 +3721,218 @@ The returned `FacebookProviderData` will include:
...
+Creating a LinkedIn Directory is very similar to [creating a directory](#create-a-directory) within Stormpath.  For a LinkedIn Directory to be configured correctly, you must specify the correct `Provider` information. 
+
+**Example Request**
+
+    Directory directory = client.instantiate(Directory.class);
+    directory.setName("my-linkedin-directory")
+             .setDescription("A LinkedIn directory");
+
+    CreateDirectoryRequest request = Directories.newCreateRequestFor(directory).
+                forProvider(Providers.LINKEDIN.builder()
```
Comment (maranai) [inline source/java/product-guide/index.markdown]: Same here. Moving the "." before forProvider might make code following easier.

File: source/java/product-guide/index.markdown [hunk 4]
```diff
@@ -3721,6 +3721,218 @@ The returned `FacebookProviderData` will include:
...
+
+After the LinkedIn Directory has been created, it needs to be [mapped with an application as an account store](#application-account-store-mappings). The LinkedIn Directory cannot be a default account store or a default group store.  Once the directory is mapped to an account store for an application, you are ready to access `Accounts` with LinkedIn User Access Tokens.
+
+### Accessing Accounts with LinkedIn User Access Tokens
+
+To access or create an account in an already created LinkedIn Directory, it is required to gather the `User Access Token` on behalf of the user.
+
+Once the Authorization Code is gathered, you can get or create the `Account` by means of the `Application` and specifying its `ProviderData`. The following example shows how you use `ProviderData` to get an `Account` for a given authorization code:
+
+
```
Comment (maranai) [inline source/java/product-guide/index.markdown]: Extra line here can be removed.

**Comment by maranai** [comment]
@mrioan I'm done with review. Only a couple of minor recommendations but otherwise consider this approved.

File: source/java/product-guide/index.markdown (hunk 1 above)
Comment (mrioan) [inline source/java/product-guide/index.markdown]: Should have been "Stormpath can leverage their access tokens"... Good catch, thanks!

File: source/java/product-guide/index.markdown (hunk 2 above)
Comment (mrioan) [inline source/java/product-guide/index.markdown]: Done

**Pull request merged**